src/storage/v2/inmemory/storage.cpp

```diff
@@ -1287,6 +1295,23 @@ void InMemoryStorage::InMemoryAccessor::Abort() {
                 // redundant. Also, `Edge/DELETE_OBJECT` isn't available when edge
                 // properties are disabled.
                 storage_->edge_count_.fetch_add(-1, std::memory_order_acq_rel);
+
+                if (!FLAGS_storage_properties_on_edges) break;
+                if (std::binary_search(index_stats.edge_type.begin(), index_stats.edge_type.end(),
```

Comment:
Where is the guarantee that index_stats.edge_type is sorted?

Reply:
When populating `index_stats.edge_type`, we go through the index, which is a `std::map<EdgeTypeId, utils::SkipList<Entry>>`, and is sorted.

Reply:
You need to do something which either documents or enforces that incidental relationship.
You want to prevent someone accidentally changing that map into an unordered_map